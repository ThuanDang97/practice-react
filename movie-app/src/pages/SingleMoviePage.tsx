import { useParams } from 'react-router-dom';
import Movie from '../components/Movie/Movie';
import { API_KEY } from '../constants/ApiKey';
import { fetchAPI } from '../helper';

const SingleMoviePage = (): JSX.Element => {
  const { id } = useParams();
  const MOVIE_URL = `https://api.themoviedb.org/3/movie/${id}?api_key=${API_KEY}&language=en-US`;
  const movieOriginal = fetchAPI(MOVIE_URL);

  if (movieOriginal) {
    return <Movie movie={movieOriginal} />;
  }

  return <div>loading</div>;
};

export default SingleMoviePage;
