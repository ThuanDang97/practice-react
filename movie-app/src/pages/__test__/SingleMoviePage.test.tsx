import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import SingleMoviePage from '../SingleMoviePage';

describe('Test Single Movie Page', () => {
  test('Single Movie Page matches DOM Snapshot', () => {
    const moviePage = (
      <Router>
        <SingleMoviePage />
      </Router>
    );
    render(moviePage);

    expect(moviePage).toMatchSnapshot();
  });

  test('Single Movie Page should render Movie', () => {
    const moviePage = (
      <Router>
        <SingleMoviePage />
      </Router>
    );
    render(moviePage);

    expect(moviePage).toBeTruthy();
  });
});
