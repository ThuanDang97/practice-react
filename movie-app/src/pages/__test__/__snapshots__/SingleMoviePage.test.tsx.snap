// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Single Movie Page Single Movie Page matches DOM Snapshot 1`] = `
<BrowserRouter>
  <SingleMoviePage />
</BrowserRouter>
`;
